src/core/app.lua

```diff
@@ -27,6 +28,36 @@ link_table, link_array = {}, {}
 
 configuration = config.new()
 
+-- Ingress packet drop monitor.
+ingress_drop_monitor = {
+   threshold = 100000,
+   wait = 20,
+   last_flush = 0,
+   last_value = ffi.new('uint64_t[1]'),
+   current_value = ffi.new('uint64_t[1]')
+}
+
+function ingress_drop_monitor:sample()
+   local sum = self.current_value
+   sum[0] = 0
+   for i = 1, #app_array do
+      local app = app_array[i]
+      if app.ingress_packet_drops and not app.dead then
+         sum[0] = sum[0] + app:ingress_packet_drops()
```

Comment:
App methods (except `new`) need to be called in protected mode using `with_restart`, otherwise a single faulty app can bring down the whole process. See how pull/push are called.